### DISCUSSION TITLE
go-cqhttp的上报消息中raw message 和message的内容有什么区别吗

### DISCUSSION THREAD

**Discussion by Unknown-Chinese-User** [Mrs4s/go-cqhttp#2371] [Q&A]
在我自己测试的时候两个字符串的内容都是一样的，会在什么情况下出现不一样，看了文档看的不大明白

我是http形式上报的，在config.yml中上报数据类型是string；我测试在config.yml中将上报数据类型改成array上报时就会和文档里讲的一样是消息链结构了，是不是可以认为当上报数据类型为string时raw_message 和message就是一样的？

**Comment by 1umine** [comment]
raw_message 是原本消息内容的字符串表示，message 则会根据配置的 string 或 array 上报格式表现不同的类型。
详见 https://github.com/botuniverse/onebot-11/blob/master/message/README.md
